tools/reindex.js

```diff
@@ -19,6 +19,12 @@ function runIndexer(indexDefinition, alias, indexer) {
   return Promise.resolve()
     .then(() => client.indices.create(mapping))
     .then(() => indexer(index))
+    .catch((err) => {
+      const errors = [].concat(err);
+      errors.map((error) => console.error(error));
+      removeIndexes([index])
+        .then(() => process.exit(-1));
+    })
```

Comment:
The `.catch()` should be the last thing we do in the Promise chain, so it catches all errors. We can't simply move this `.catch()` as is, though, as if the error happened during `updateAlias()`, we don't want to remove the indexes. This gives us two options: don't remove the indexes (they'll be removed on the next reindex run), or use two `.catch()`. We could change it to something like:

```python
    # ...
    .then(() => indexer(index))
    .catch((err) => removeIndexes([index]).then(() => { throw err }))
    .then(() => updateAlias(index, alias))
    .catch((err) => {
      console.error(err);
      process.exit(-1);
    });
```

This way, if the error was on the `indexer()` call or before, we try to remove the newly created index and rethrow the original error. It'll go down the chain, skipping all `.then()` (i.e. the `updateAlias()`) and end up in the last `.catch()`, where we print the error and exit. If the error is in `updateAlias()`, we just print the error and exist (i.e. don't remove the index).

Note that we're returning the `removeIndexes()`. This is necessary because `removeIndexes()` returns a promise, which needs to be part of the chain to be guaranteed to run.